modules/dxp/apps/search-experiences/search-experiences-web/src/main/java/com/liferay/search/experiences/web/internal/blueprint/admin/portlet/action/GetSitesMVCResourceCommand.java

```diff
@@ -122,10 +122,29 @@ protected JSONObject getSitesJSONObject(
 				return 0;
 			});
 
+		int pageSize = ParamUtil.getInteger(resourceRequest, "pageSize");
+		int totalCount = allGroups.size();
+
+		int lastPage = (int)Math.ceil((double)totalCount / pageSize);
+
+		int page = ParamUtil.getInteger(resourceRequest, "page");
+
+		if (page > lastPage) {
+			page = lastPage;
+		}
+
+		int endPosition = pageSize * page;
+
+		int startPosition = endPosition - pageSize;
+
```

Comment:
Naming would be better as:
pageSize -> delta
totalCount -> total
endPostition -> end
startPosition -> start

Reply:
pageSize, page, and totalCount seem to be standard in the headless APIs